### PR TITLE
FIx rails-app and document how to investigate the server

### PR DESCRIPTION
**Pull request by pitr-ch**

### CODE REVIEW

File: doc/contributor/quick-rails-testing.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+# Quick rails testing
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "rails" -> "Rails" for consistency.

File: doc/contributor/quick-rails-testing.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+This repository contains small Rails application for testing
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "contains small" -> "contains a small"

File: doc/contributor/quick-rails-testing.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
+# Quick rails testing
+
+This repository contains small Rails application for testing 
+in `test/truffle/ecosystem/rails-app` directory.
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "in" -> "in the"

File: doc/contributor/quick-rails-testing.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
+# Quick rails testing
+
+This repository contains small Rails application for testing 
+in `test/truffle/ecosystem/rails-app` directory. 
+The test can be executed with `jt test ecosystem rails-app` in TruffleRuby home directory.
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "in TruffleRuby" -> "in TruffleRuby's" or "in the TruffleRuby"

File: doc/contributor/quick-rails-testing.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+When executed it prepares its environment, starts rails server,
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "starts rails" -> "starts the rails"

File: doc/contributor/quick-rails-testing.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
+# Quick rails testing
+
+This repository contains small Rails application for testing 
+in `test/truffle/ecosystem/rails-app` directory. 
+The test can be executed with `jt test ecosystem rails-app` in TruffleRuby home directory.
+When executed it prepares its environment, starts rails server, 
+runs few tests against JSON API, and shut downs the rails server.
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "runs few" -> "runs a few"

File: doc/contributor/quick-rails-testing.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+It requires Redis as a database to by running in default configuration on `localhost:6379`.
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "running in" -> "running in its"

File: doc/contributor/quick-rails-testing.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+The `jt test ecosystem rails-app` should be executed in truffleruby home,
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "in truffleruby" -> "in TruffleRuby's" or "in the TruffleRuby" (match this to my previous comment on the phrasing).

File: doc/contributor/quick-rails-testing.md [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+This repository contains small Rails application for testing 
+in `test/truffle/ecosystem/rails-app` directory. 
+The test can be executed with `jt test ecosystem rails-app` in TruffleRuby home directory.
+When executed it prepares its environment, starts rails server, 
+runs few tests against JSON API, and shut downs the rails server.
+It requires Redis as a database to by running in default configuration on `localhost:6379`.
+
+The `jt test ecosystem rails-app` should be executed in truffleruby home, 
+the output will be similar to following:
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "home, the" -> "home. The"

File: doc/contributor/quick-rails-testing.md [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+
+Started DELETE "/people/destroy_all.json" for ::1 at 2018-01-17 09:02:51 +0100
+Processing by PeopleController#destroy_all as JSON
+Completed 200 OK in 16ms (Views: 1.4ms)
+true+ kill %1
+++ cat tmp/pids/server.pid
++ kill 9617
+```
+
+There is several levels of indirection but it always prints what command is actually
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "There is" -> "There are"

File: doc/contributor/quick-rails-testing.md [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+Started DELETE "/people/destroy_all.json" for ::1 at 2018-01-17 09:02:51 +0100
+Processing by PeopleController#destroy_all as JSON
+Completed 200 OK in 16ms (Views: 1.4ms)
+true+ kill %1
+++ cat tmp/pids/server.pid
++ kill 9617
+```
+
+There is several levels of indirection but it always prints what command is actually 
+executed, it's prefixed with `+` if it's executed by bash or by `$` if it's our tool.
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "executed, it's" -> "executed. It's"

File: doc/contributor/quick-rails-testing.md [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+
+If the test finishes without problem then the rails server can be executed from
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "from" -> "from the"

File: doc/contributor/quick-rails-testing.md [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+`path/to/truffleruby/test/truffle/ecosystem/rails-app` directory for further investigation.
+The server can be started with following command which can be copied out from the output.
+
+```bash
+TRUFFLERUBY_RESILIENT_GEM_HOME="" GEM_HOME="path/to/truffleruby/truffleruby-gem-test-pack-5/gems" GEM_PATH="path/to/truffleruby/truffleruby-gem-test-pack-5/gems" PATH="path/to/truffleruby/truffleruby-gem-test-pack-5/gems/bin:<snipped>" NO_FORK="true" path/to/truffleruby/bin/truffleruby -J-Xmx2G -J-ea -J-esa -Xcore.load_path\=path/to/truffleruby/src/main/ruby -I path/to/truffleruby/test/truffle/ecosystem/rails-app/.truffleruby-tool_bundle/mocks -r openssl-stubs -r stubs -S bundle exec bin/rails server
+```
+
+The command can be further modified for inspection.
+First it's useful to use jt not to run truffleruby directly.
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "truffleruby" -> "TruffleRuby"

File: doc/contributor/quick-rails-testing.md [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+
+The command can be further modified for inspection.
+First it's useful to use jt not to run truffleruby directly. 
+(`-Xcore.load_path` option can be then omitted.)
+
+```bash
+env TRUFFLERUBY_RESILIENT_GEM_HOME="" GEM_HOME="/Users/pitr/Workspace/labs/truffleruby-ws/truffleruby/truffleruby-gem-test-pack-5/gems" GEM_PATH="/Users/pitr/Workspace/labs/truffleruby-ws/truffleruby/truffleruby-gem-test-pack-5/gems" PATH="path/to/truffleruby/truffleruby-gem-test-pack-5/gems/bin:<snipped>" NO_FORK="true" ../../../../tool/jt.rb ruby -J-Xmx2G -J-ea -J-esa -I /Users/pitr/Workspace/labs/truffleruby-ws/truffleruby/test/truffle/ecosystem/rails-app/.truffleruby-tool_bundle/mocks -r openssl-stubs -r stubs -S bundle exec ./bin/rails server
+```
+
+To run on graal pass `--graal` to `jt ruby`.
```
Comment (nirvdrum) [inline doc/contributor/quick-rails-testing.md]: "graal" -> "Graal"

**Comment by pitr-ch** [comment]
Thanks, and sorry it looks like I didn't read it after myself even once this time, I hope I would have caught at least some of them.

File: doc/contributor/quick-rails-testing.md [hunk 15]
```diff
@@ -0,0 +1,183 @@
+# Quick Rails testing
+
+This repository contains a small Rails application for testing 
+in the `test/truffle/ecosystem/rails-app` directory. 
+The test can be executed with `jt test ecosystem rails-app` in TruffleRuby's home directory.
+When executed it prepares its environment, starts the Rails server, 
+runs a few tests against JSON API, and shut downs the rails server.
+It requires Redis as a database to by running in its default configuration on `localhost:6379`.
```
Comment (eregon) [inline doc/contributor/quick-rails-testing.md]: *be* running

**Pull request merged**